File: backend/models/Dog.js
const mongoose = require('mongoose')
const Schema = mongoose.Schema;

const DogSchema = new Schema({
    userId: { type: String, required: true },
    dogName: { type: String, required: true },
    dogAge: { type: String, required: true },
    dogBreed: { type: String, required: true },
    dogHobby: { type: String, required: false, default: "What's your dog's hobbies?" },
    dogPersonality: { type: String, required: false, default: "What's your dog's personality?" },
    createDate: { type: Date, default: Date.now }
})

const DogModel = mongoose.model("Dog", DogSchema);

module.exports = DogModel

File: lib/models/dog.dart
import '../utils/mobile_library.dart';

class Dog extends ChangeNotifier {
  String _id;
  String userId;
  String dogName;
  String dogAge;
  String dogBreed;
  String dogHobby;
  String dogPersonality;

  Dog(this._id, this.userId, this.dogName, this.dogAge, this.dogBreed, this.dogHobby, this.dogPersonality);

  factory Dog.fromJson(dynamic json) {
    return Dog(
        json['_id'] as String,
        json['userId'] as String,
        json['dogName'] as String,
        json['dogAge'] as String,
        json['dogBreed'] as String,
        json['dogHobby'] as String,
        json['dogPersonality'] as String
    );
  }


  String get id => _id;

  @override
  String toString() {
    return 'Dog{_id: $_id, userId: $userId, dogName: $dogName, dogAge: $dogAge, dogBreed: $dogBreed, dogHobby: $dogHobby, dogPersonality: $dogPersonality}';
  }
}